### PR TITLE
Test commander change duration

### PR DESCRIPTION
**Pull request by karottenreibe**
allow changing duration of manual tests recorded with the test commander

### CODE REVIEW

**Review by karottenreibe — COMMENTED** [review]
@Raphael-N here's two suggestions for your code :-)

**Comment by karottenreibe** [comment]
@Raphael-N I tested our changes and they work as expected with Blogifier :-)

**Comment by Raphael-N** [comment]
I implemented your comments. The rest looks good to me. Can be merged from my side.

File: UploadDaemon/Scanning/TraceFile.cs [hunk 1]
```diff
@@ -21,7 +21,7 @@ public class TraceFile
         private static readonly Regex ProcessLineRegex = new Regex(@"^Process=(.*)", RegexOptions.IgnoreCase);
         private static readonly Regex AssemblyLineRegex = new Regex(@"^Assembly=([^:]+):(\d+)");
         private static readonly Regex CoverageLineRegex = new Regex(@"^(?:Inlined|Jitted|Called)=(\d+):(?:\d+:)?(\d+)");
-        private static readonly Regex TestCaseLineRegex = new Regex(@"^(?:Test)=((?:Start|End)):([^:]+):(.+)");
+        private static readonly Regex TestCaseLineRegex = new Regex(@"^(?:Test)=((?:Start|End)):([^:]+):([^:]+)(?::(\d+))?");
```
[the remark this answers is not in the document]
Comment (Raphael-N) [reply UploadDaemon/Scanning/TraceFile.cs]: Good idea. Also did it for some of the other regexes.

File: UploadDaemon/Scanning/TraceFile.cs [hunk 2]
```diff
@@ -70,6 +70,7 @@ public ICoverageReport ToReport(Func<Trace, SimpleCoverageReport> traceResolver)
             DateTime currentTestStart = default;
             Trace currentTestTrace = noTestTrace;
             DateTime currentTestEnd;
+            long testDuration;
```
[the remark this answers is not in the document]
Comment (Raphael-N) [reply UploadDaemon/Scanning/TraceFile.cs]: 👍

File: UploadDaemon/Scanning/TraceFile.cs [hunk 3]
```diff
@@ -20,8 +20,8 @@ public class TraceFile
         private static readonly Regex TraceFileRegex = new Regex(@"^coverage_\d*_\d*.txt$");
         private static readonly Regex ProcessLineRegex = new Regex(@"^Process=(.*)", RegexOptions.IgnoreCase);
         private static readonly Regex AssemblyLineRegex = new Regex(@"^Assembly=([^:]+):(\d+)");
-        private static readonly Regex CoverageLineRegex = new Regex(@"^(?:Inlined|Jitted|Called)=(\d+):(?:\d+:)?(\d+)");
-        private static readonly Regex TestCaseLineRegex = new Regex(@"^(?:Test)=((?:Start|End)):([^:]+):([^:]+)(?::(\d+))?");
+        private static readonly Regex CoverageLineRegex = new Regex(@"^(?:Inlined|Jitted|Called)=(?<assembly>\d+):(?:\d+:)?(?<functionToken>\d+)");
+        private static readonly Regex TestCaseLineRegex = new Regex(@"^(?:Test)=((?:Start|End)):(?<date>[^:]+):(?<testname>[^:]+)(?::(?<duration>\d+))?");
```
[the remark this answers is not in the document]
Comment (Raphael-N) [reply UploadDaemon/Scanning/TraceFile.cs]: Done 🙂

**Comment by karottenreibe** [comment]
One more minor comment @Raphael-N 
I also did another test run, still works with Blogifier

**Pull request merged**